### ISSUE TITLE
Trust ~/.local/share/applications/*.desktop files as default

### ISSUE THREAD

**Issue by stefonarch** [lxqt/pcmanfm-qt#1519]
I see no reason why not.

##### Current Behavior

Exclamation marks all over.

##### Context

https://blog.lxde.org/2021/02/19/libfm-pcmanfm-1-3-2-released/
##### System Information

* PcManFm-qt version: latest git, 1.0.0

**Comment by tsujan** [comment]
> Exclamation marks all over.

The trust emblem isn't shown inside `QStandardPaths::ApplicationsLocation`, which includes `~/.local/share/applications` and `/usr/share/applications`. I wonder how you see exclamation marks there.

As for trusting, the items of these folders are for menus, not for being clicked and executed automatically, unless the user explicitly chooses so.

Desktop entries should be automatically trusted only when they are created by using pcmanfm-qt itself (through Create Launcher).

**Comment by stefonarch** [comment]
I remembered wrong, thought I'd trusted them manually, it's already like that.

**Comment by tsujan** [comment]
I remember I discussed it with Alf. Here I explain it in more detail.

If you click some items, unpredictable processes might start to run and remain in the background — for example, no one wants to have the CPU-unfriendly GTK3/4 tracker running in the background of his LXQt session, and some processes might even be incompatible with the current session. There are also desktop entries in those folders that contain `NoDisplay=true`, etc. Krita has 23 items there! And so on.

In short, the items of those folders aren't all safe for being clicked inside a file manager. So, the user should be warned, unless he chooses otherwise.

However, you shouldn't see the "trust emblem" in those folders. The emblem should appear in their subfolders though, because they aren't among the standard app paths.

**Comment by stefonarch** [comment]
Now I remember where I hit ctrl+A and checked "trust": it was in` ~/.local/share/file-manager/actions/`. 
I tested a new file, it has exclamation marks but with right click on an item and selecting it it executes. 

What about adding this directory instead?
Other thing I saw is that the "trust" checkbox is unchecked, it has no meaning there, same in /usr/share/applications.

**Comment by tsujan** [comment]
> Other thing I saw is that the "trust" checkbox is unchecked - it has no meaning in ~/.local/share/applications/, same in /usr/share/applications.

There should be a miscommunication somewhere. Sorry if it's on my part! I explained above why desktop entries in those folders should *not* be trusted automatically. *Only the exclamation icon isn't shown there* because so many emblems would be useless with standard paths.

`~/.local/share/file-manager/actions/` isn't a standard path.

**Comment by tsujan** [comment]
Please also note that the absence of exclamation emblem isn't *equal* to trustworthiness. All executables in `/usr/bin/` are untrusted, yet none of them has an emblem. We use the emblem only for *desktop entries* outside the standard app paths.

**Comment by stefonarch** [comment]
I got it. Better wording: don't show exclamation marks  for custom action files. And maybe "disable checkbox where it cannot work (/)".

**Comment by tsujan** [comment]
> And maybe "disable checkbox where it cannot work

The checkbox does work everywhere it appears. Its real job is that the executable/desktop entry will be executed without prompt if it's checked. In the case of desktop entries with the exclamation emblem, it also removes the emblem, as a byproduct.

> don't show exclamation marks for custom action files.

As said above, that folder isn't a standard path. Making an exception for it would be too arbitrary.

**Comment by stefonarch** [comment]
> The checkbox does work everywhere it appears.

I missed that, true.

**Comment by tsujan** [comment]
What you want is quite possible. The problem is where we should stop making exceptions. I made an exception for `QStandardPaths::ApplicationsLocation` because I found too many exclamation emblems useless, and other file managers did it too, as far as I remember. The good thing about `QStandardPaths::ApplicationsLocation` is that it's clean; we don't need to find and type the paths in the code.
